### PR TITLE
REST API: Add "Create User" endpoint

### PR DESCRIPTION
**Pull request by lezama**

### CODE REVIEW

File: json-endpoints/jetpack/class.jetpack-json-api-site-user-new-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+					break;
+				case 'login':
+					$this->user[ 'user_login' ] = $value;
+					break;
+				case 'email':
+					$this->user[ 'user_email' ] = $value;
+					break;
+				case 'display_name':
+				case 'name':
+					$this->user[ 'last_name' ] = $value;
```
Comment (roccotripaldi) [inline json-endpoints/jetpack/class.jetpack-json-api-site-user-new-endpoint.php]: last_name = display_name

File: json-endpoints/jetpack/class.jetpack-json-api-site-user-new-endpoint.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			return new WP_Error( 'Create User Error', 'user_login and user_email can\'t be empty', 400 );
+		}
+		if ( is_multisite() ) {
+			$result = wpmu_validate_user_signup( $this->user[ 'user_login' ], $this->user[ 'user_email' ] );
+			if ( is_wp_error( $result['errors'] ) && ! empty( $result['errors']->errors ) ) {
+				return $result['errors'];
+			}
+		}
+		if ( empty( $this->user['user_email'] ) ){
+			$this->user['user_email'] = wp_generate_password();
```
Comment (roccotripaldi) [inline json-endpoints/jetpack/class.jetpack-json-api-site-user-new-endpoint.php]: user_email = user_pass

**Comment by jeherve** [comment]
@lezama Is this PR still valid and being worked on?

**Comment by lezama** [comment]
let's keep it open

cc @ebinnion

**Comment by ebinnion** [comment]
I don't anticipate that we'll be using this endpoint soon. We will be working on an endpoint to accept WP.com invitations and create users, but I'm not sure that it'll use this code.

And with wp-api work, we may just end up using a core implementation for creating users.

I'm going to lean towards closing this PR for now. We can always re-open and restore the branch later if we need to get at the code.

**Pull request closed without merging**